src/api/spec/models/attrib_spec.rb

```diff
@@ -1,19 +1,106 @@
 require 'rails_helper'
 
-RSpec.describe Attrib do
+RSpec.describe Attrib, :type => :model do
   let(:attribute) { create(:attrib, project: create(:project)) }
   let(:project) { create(:project) }
   let(:package) { create(:package) }
 
+  context "#fullname" do
+    it { expect(attribute.fullname).to eq("#{attribute.namespace}:#{attribute.name}") }
+  end
+
+  context "#container" do
+    context "attribute with project" do
+      it { expect(attribute.container).to eq(attribute.project) }
+    end
+
+    context "attribute with package" do
+      let(:attribute_with_package) { create(:attrib, package: package) }
+
+      it { expect(attribute_with_package.container).to eq(package) }
+    end
+  end
+
   context "#container=" do
-    it "sets the project" do
-      attribute.container = project
-      expect(attribute.container).to be(project)
+    context "assigning a project" do
+      before do
+        attribute.container = project
+      end
+
+      it { expect(attribute.container).to be(project) }
+      it { expect(attribute.project).to be(project) }
+      it { expect(attribute.package).to be_nil }
+
+      context "and then assigning a package" do
```

Comment:
> In your gist code I don't see a test of comparing the values container, project and package methods return after an assignment of a package and then a project. Should I drop that test then?

I would say yes because they are two different methods, so trying to test two methods in the same describe block makes leaves us with too many scenarios to test.